dali/pipeline/operators/geometric/flip.cu

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cuda_runtime_api.h>
+#include "dali/pipeline/operators/geometric/flip.h"
+
+namespace dali {
+
+template <>
+Flip<GPUBackend>::Flip(const OpSpec &spec) : Operator<GPUBackend>(spec), spec_(spec) {}
+
+template <bool flipX, bool flipY, typename T>
+__global__ void FlipKernel(T *__restrict__ output, const T *__restrict__ input, size_t layers,
+                           size_t height, size_t width, size_t channels_per_layer) {
```

Comment:
I would say, that writing this function in proper way is to create a seperate `__host__ __device__ FlipKernelImpl()`, which would have unit tests written. Then `__global__ FlipKernel` function would be cuda-configured and provide only execution params for `FlipKernelImpl`. Cause in current situation, actual flip implementation is tested as far as the operator execution, with bunch of different things happening along the way...

Reply:
In the new kernel api approach, kernels have their own tests